### PR TITLE
Conditions to transfer on button press on button event

### PR DESCRIPTION
**Pull request by OHerasym**
### PR Changes:
 -  Fix OnButtonPress & OnButtonEvent description
  -  Change conditions to transfer OnButtonEvent
   - Change conditions to transfer OnButtonPress

Related: [APPLINK-25470](https://adc.luxoft.com/jira/browse/APPLINK-25470)

Old [pull request](https://github.com/smartdevicelink/sdl_core/pull/757)

### CODE REVIEW

**Comment by OHerasym** [comment]
Please review: @dtrunov , @anosach-luxoft , @LuxoftAKutsan ,@VVeremjova, @orhan-mehmedov

**Comment by OHerasym** [comment]
@TMelnyk, Please review HMI_API.xml file

**Comment by TMelnyk** [comment]
reviewed

**Comment by dtrunov** [comment]
@OHerasym Reviewed

File: src/components/application_manager/src/commands/mobile/on_button_press_notification.cc [hunk 1]
```diff
@@ -106,20 +106,22 @@ void OnButtonPressNotification::Run() {
...
-    if ((static_cast<uint32_t>(mobile_apis::ButtonName::OK) == btn_id) &&
-        (mobile_api::HMILevel::HMI_FULL != subscribed_app->hmi_level())) {
-      continue;
+    // if "appID" is present  send it to Named app , only if its FULL or
+    // LIMITED
+    if (app && app->app_id()) {
+      SendButtonPress(subscribed_app);
+    } else if (subscribed_app->IsFullscreen()) {
+      // if No "appID" - send it FULL apps only.
+      SendButtonPress(subscribed_app);
```
Comment (orhan-mehmedov) [inline src/components/application_manager/src/commands/mobile/on_button_press_notification.cc]: @OHerasym, seems there are same actions in "if" and in "else if". Can it be just one "if"? (is there important side effects in IsFullscreen?)

File: src/components/application_manager/src/commands/mobile/on_button_event_notification.cc [hunk 2]
```diff
@@ -106,20 +106,21 @@ void OnButtonEventNotification::Run() {
...
       continue;
     }
 
-    //Send ButtonEvent notification for OK button only in HMI_FULL mode
-    if ((static_cast<uint32_t>(mobile_apis::ButtonName::OK) == btn_id) &&
-        (!subscribed_app->IsFullscreen())) {
-      continue;
+    // if "appID" is present  send it to Named app , only if its FULL or
+    // LIMITED
+    if (app && app->app_id()) {
```
Comment (anosach-luxoft) [inline src/components/application_manager/src/commands/mobile/on_button_event_notification.cc]: @OHerasym I don't like these complicated `if` statements. I think this checking could be performed in previous `if`.

File: src/components/application_manager/src/commands/mobile/on_button_press_notification.cc (hunk 1 above)
Comment (OHerasym) [inline src/components/application_manager/src/commands/mobile/on_button_press_notification.cc]: @orhan-mehmedov, 7073b13, Done

File: src/components/application_manager/src/commands/mobile/on_button_event_notification.cc (hunk 2 above)
Comment (OHerasym) [inline src/components/application_manager/src/commands/mobile/on_button_event_notification.cc]: @orhan-mehmedov, 7073b13, Done

**Comment by orhan-mehmedov** [comment]
@OHerasym Reviewed

File: src/components/application_manager/src/commands/mobile/on_button_event_notification.cc [hunk 3]
```diff
@@ -57,6 +57,9 @@ void OnButtonEventNotification::Run() {
       static_cast<uint32_t>(
           (*message_)[strings::msg_params][hmi_response::button_name].asInt());
 
+  ApplicationSharedPtr app = ApplicationManagerImpl::instance()->application(
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/mobile/on_button_event_notification.cc]: @OHerasym can app be const?

File: src/components/application_manager/src/commands/mobile/on_button_event_notification.cc [hunk 4]
```diff
@@ -57,6 +57,9 @@ void OnButtonEventNotification::Run() {
...
+      (*message_)[strings::msg_params][strings::app_id].asUInt());
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/mobile/on_button_event_notification.cc]: @OHerasym is app-id mandatory parameter according API? if not, please check if it is exists.

File: src/components/application_manager/src/commands/mobile/on_button_event_notification.cc [hunk 5]
```diff
@@ -106,20 +106,15 @@ void OnButtonEventNotification::Run() {
       continue;
     }
 
-    //Send ButtonEvent notification only in HMI_FULL or HMI_LIMITED mode
-    if ((mobile_api::HMILevel::HMI_FULL != subscribed_app->hmi_level()) &&
-        (mobile_api::HMILevel::HMI_LIMITED != subscribed_app->hmi_level())) {
-      LOG4CXX_WARN_EXT(logger_, "OnButtonEvent in HMI_BACKGROUND or NONE");
-      continue;
+    if (app || subscribed_app->IsFullscreen()) {
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/mobile/on_button_event_notification.cc]: @OHerasym You check app pointer but does not use it inside if. looks confusing.

File: src/components/application_manager/src/commands/mobile/on_button_press_notification.cc [hunk 6]
```diff
@@ -57,6 +57,9 @@ void OnButtonPressNotification::Run() {
       static_cast<uint32_t>(
           (*message_)[strings::msg_params][hmi_response::button_name].asInt());
 
+  ApplicationSharedPtr app = ApplicationManagerImpl::instance()->application(
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/mobile/on_button_press_notification.cc]: @OHerasym can app be const?

File: src/components/application_manager/src/commands/mobile/on_button_event_notification.cc (hunk 3 above)
Comment (OHerasym) [inline src/components/application_manager/src/commands/mobile/on_button_event_notification.cc]: @LuxoftAKutsan , b4e3de2 , Done

File: src/components/application_manager/src/commands/mobile/on_button_event_notification.cc (hunk 4 above)
Comment (OHerasym) [inline src/components/application_manager/src/commands/mobile/on_button_event_notification.cc]: @LuxoftAKutsan , there is logic for case when app_id is absent

File: src/components/application_manager/src/commands/mobile/on_button_event_notification.cc (hunk 5 above)
Comment (OHerasym) [inline src/components/application_manager/src/commands/mobile/on_button_event_notification.cc]: @LuxoftAKutsan , b4e3de2 , Done , added comment

File: src/components/application_manager/src/commands/mobile/on_button_press_notification.cc (hunk 6 above)
Comment (OHerasym) [inline src/components/application_manager/src/commands/mobile/on_button_press_notification.cc]: @LuxoftAKutsan , b4e3de2 , Done

**Comment by LuxoftAKutsan** [comment]
Reviewed

**Comment by anosach-luxoft** [comment]
Reviewed

**Pull request merged**